### PR TITLE
Add OpenOrb

### PR DESCRIPTION
**Pull request by mjuric**
Add packaging for OpenOrb, an open-source orbit-computation package (https://github.com/oorb/oorb). This PR adds openorb and the related data packages (built as noarch).

OpenOrb does not build on Windows (only Linux and macOS are supported).

### CODE REVIEW

**Comment by mjuric** [comment]
The CI builds pass; this PR is ready to be looked at (and hopefully, merged).

**Comment by mjuric** [comment]
Hi @conda-forge/staged-recipes, this PR now passes all checks and is ready to be reviewed.

File: recipes/openorb-data/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{% set name = "openorb-data" %}
+{% set version = "1.0.1.20181028" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
```
Comment (pkgw) [inline recipes/openorb-data/meta.yaml]: Should this URL now have `/oorb/` instead of `/mjuric/`?

File: recipes/openorb-eph-builder/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{% set name = "openorb-eph-builder" %}
+{% set version = "1.0.1.20181028" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/mjuric/oorb.git
```
Comment (pkgw) [inline recipes/openorb-eph-builder/meta.yaml]: Ditto — `/oorb/` instead of `/mjuric/`?

File: recipes/openorb/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{% set name = "openorb" %}
+{% set version = "1.0.1.20181028" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/mjuric/oorb.git
```
Comment (pkgw) [inline recipes/openorb/meta.yaml]: One last time — `/oorb/`?

File: recipes/openorb-eph-builder/build.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# conda's gfortran on Linux does not install a binary named 'gfortran'
+echo "gfortran = $(which gfortran)"
+[[ ! -f "$BUILD_PREFIX/bin/gfortran" && ! -z "$GFORTRAN" ]] && ln -s "$GFORTRAN" "$BUILD_PREFIX/bin/gfortran"
+echo "gfortran = $(which gfortran)"
```
Comment (pkgw) [inline recipes/openorb-eph-builder/build.sh]: conda-forge is transitioning the way in which it deals with compilers. In the near future, build scripts should always use `$FC` as their FORTRAN compiler. But the older compiler packages don't set this variable. Would a construct like this work with your configure script?

```
if [ -z "$FC" ] ; then
    export FC=gfortran
fi

./configure "$FC" opt --prefix="$PREFIX"
```

?

File: recipes/openorb/build.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# conda's gfortran on Linux does not install a binary named 'gfortran'
+echo "gfortran = $(which gfortran)"
+[[ ! -f "$BUILD_PREFIX/bin/gfortran" && ! -z "$GFORTRAN" ]] && ln -s "$GFORTRAN" "$BUILD_PREFIX/bin/gfortran"
+echo "gfortran = $(which gfortran)"
```
Comment (pkgw) [inline recipes/openorb/build.sh]: Ditto w.r.t `$FC`.

File: recipes/openorb-data/meta.yaml (hunk 1 above)
Comment (mjuric) [reply recipes/openorb-data/meta.yaml]: I'm building from my fork for now as I have a number of build system-related patches there that are destined for oorb/oorb but haven't been merged yet (https://github.com/oorb/oorb/pull/63). I thought about moving them all into patches (or one omnibus patch), but this seemed cleaner for now (the patch would be heavy). That said, happy to do it either way!

(again, I plan to switch to building from `oorb/oorb` as soon as the PR is merged).

File: recipes/openorb-eph-builder/build.sh (hunk 4 above)
Comment (mjuric) [reply recipes/openorb-eph-builder/build.sh]: Yes, I'm also unhappy with the present hack; the issue is that the upstream code looks for a compiler by name (i.e., its `configure` script generates a `Makefile.include` that says `FC=gfortran`).

The long-term solution is to modernize oorb's build system, after which this hack won't be necessary. On my TODO list.

**Pull request merged**